### PR TITLE
Cross-platform `which` to find nodejs & NPM

### PR DESCRIPTION
**Pull request by john-osullivan**
This is the PR including the code from https://github.com/shakacode/react_on_rails/issues/373, added in a link to the source in case somebody from the future is curious.  All it does is replace `install-generator.rb`'s usage of the Unix-only `which` command with a cross-platform method written by a friendly StackOverflow user.

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/374)

### CODE REVIEW

**Comment by justin808** [comment]
@john-osullivan You'll need to get this to pass CI.

**Comment by justin808** [comment]
@john-osullivan Can you please rebase on top of master and get this to pass CI.

**Comment by jbhatab** [comment]
@john-osullivan Can you rebase this on top of master and fix the tests?

**Comment by justin808** [comment]
@john-osullivan Please rebase on top of master and squash to one commit, and I'll merge.

**Comment by justin808** [comment]
@john-osullivan Also, please update the changelog.

**Comment by john-osullivan** [comment]
Will do!  Working on it now.

**Comment by john-osullivan** [comment]
Some of the changes from commits between my original commit and this updated one got included, which I'm not sure how to avoid, but this is now flattened into one commit and will hopefully pass Travis in just a moment!

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,12 +3,17 @@ All notable changes to this project will be documented in this file. Items under
 
 Contributors: please follow the recommendations outlined at [keepachangelog.com](http://keepachangelog.com/). Please use the existing headings and styling as a guide, and add a link for the version diff at the bottom of the file. Also, please update the `Unreleased` link to compare to the latest release version.
 ## [Unreleased]
+##### Added
+- Non-digested version of assets in public folder [#413](https://github.com/shakacode/react_on_rails/pull/413) by [alleycat-at-git]
+- Cross-platform `which` method to find node.js and NPM when running on Windows.  See [#374](https://github.com/shakacode/react_on_rails/pull/374) by [john-osullivan](https://github.com/john-osullivan/).
+
```
Comment (justin808) [inline CHANGELOG.md:9]: this is not right

this is not part of this change

**Comment by justin808** [comment]
@john-osullivan I need one commit that only shows your changes in the diffs. If you can't figure out the right sequence of commands, just make a new commit on top of master and force push to this same branch name.

**Comment by justin808** [comment]
@john-osullivan or anybody else that wants this, please open a new PR for this.
